app/javascript/mastodon/utils/notifications.ts

```diff
@@ -0,0 +1,46 @@
+// Handles browser quirks, based on
+// https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API/Using_the_Notifications_API
+
+interface WritableNotification {
+  permission: NotificationPermission;
+}
```

Comment:
Similar issue to https://github.com/mastodon/mastodon/pull/27894/files#r1396032427: TypeScript assumes a newer / more strict browser support range.

Reply:
Is all of this still needed?

We no longer support really old browsers, but I am having trouble figuring out if `Notification.permission` might be unimplemented on some modern browsers

Reply:
I haven't been able to find a situation where it isn't.